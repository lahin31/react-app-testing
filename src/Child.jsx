import React, { Component } from 'react';

class Child extends Component {
  state = { 
    todos: [],
    todo: {
      text: ""
    },
    showText: false
  }
  render() { 
    const {
      todo: {
        text
      }
    } = this.state
    const submitForm = e => {
      console.log(e)
    }

    const changeText = e => {
      let todo = {...this.state.todo};
      todo.text = e.target.value;
      this.setState({ todo })
    }

    return ( 
      <div>
        { this.state.showText ? <p className="para">I love React</p>: <p className="para">React is a Library</p> }
        <button onClick={() => {
          this.setState(prevState =>({
            showText: !prevState.showText
          }))
        }} className="show_text_btn">Show Text</button>
        <form onSubmit={submitForm}>
          <input type="text" value={text} onChange={e => changeText(e)}/>
          <button className="add_text_btn">Add</button>
        </form>
      </div>
    );
  }
}
 
export default Child;