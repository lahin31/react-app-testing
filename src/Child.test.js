/**
 * Child.test.js - simple testing file
 * tested a component is rendered successfully or not by SNAPSHOT Testing
 * tested a state value from initial period
 * tested a textContent value by using find and text
 * tested a click event using mount(which gave us the DOM Representation of the component to do test event)
 */

import Child from "./Child"

describe("Todos Component", () => {  
  it("renders correctly", () => {
    const wrapper = shallow(<Child />) // shallow - returns a light-weight representation of what component renders
    expect(wrapper).toMatchSnapshot();
  })

  it("the initial length of todos array should be zero", () => {
    const wrapper = shallow(<Child />) // 
    const countTodosLength = wrapper.state().todos.length
    expect(countTodosLength).toEqual(0)
  })

  it("text value of button tag should be Add", () => {
    const wrapper = shallow(<Child />)
    const text = wrapper.find(".add_text_btn").text()
    expect(text).toEqual("Add")
  })

  it("it triggers the text", () => {
    const wrapper = mount(<Child />) // mount - returns a full DOM representation of the component
    wrapper
      .find('.show_text_btn')
      .simulate('click')
    const para = wrapper.find('.para').text()
    expect(para).toEqual("I love React")
  })

})