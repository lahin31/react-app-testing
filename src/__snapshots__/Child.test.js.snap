// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Component renders correctly 1`] = `
<div>
  <p
    className="para"
  >
    React is a Library
  </p>
  <button
    className="show_text_btn"
    onClick={[Function]}
  >
    Show Text
  </button>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="add_text_btn"
    >
      Add
    </button>
  </form>
</div>
`;
